src/domain/errors/coded.error.ts

```diff
@@ -0,0 +1,8 @@
+export class CodedError extends Error {
+  constructor(
+    message: string,
+    public readonly code: string
```

Comment:
I'd like people's opinions on this. ATM I'm just passing in a string for the type of error it is. Should we look to use error codes such as `DBERR001` which then maps to a specific scenario?